tools/alive_parser.cpp

```diff
@@ -365,7 +366,14 @@ static unique_ptr<Instr> parse_binop(string_view name, token op_token) {
   auto &type = parse_type();
   auto &a = parse_operand(type);
   parse_comma();
-  auto &b = parse_operand(type);
+
+  // if second operand's type is not mentioned,
+  // assume type to be same as first
+  auto *rhsType = &parse_type();
+  if (!rhsType->isDefined()) {
+    rhsType = &type;
+  }
+  auto &b = parse_operand(*rhsType);
```

Comment:
Uhm, not sure I like this..  Can you factor out this change in a new function (one that takes a default type and returns that default type if it there's no type to parse)?